default/index.yaml

```diff
@@ -87,9 +87,25 @@ js:
   # Syntax highlighting using prism.js
   prism: |
     <!-- Prism.js -->
-    <link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.26.0/themes/prism-tomorrow.min.css" rel="stylesheet" />
-    <script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.26.0/prism.min.js"></script>
-    <script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.26.0/plugins/autoloader/prism-autoloader.min.js"></script>
+    <link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.28.0/themes/prism-tomorrow.min.css" rel="stylesheet" />
+    <script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.28.0/prism.min.js" data-manual></script>
+    <script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.28.0/plugins/autoloader/prism-autoloader.min.js"></script>
+    <script>
+    function highlightAllExceptMermaid() {
+      var elements = document.querySelectorAll('pre > code[class*="language"]:not(.language-mermaid)');
+      for (var element of elements) {
+        Prism.highlightElement(element, false);
+      }
+    }
+    window.isHotReload = window.isHotReload !== undefined;
+    if (window.isHotReload) {
+      highlightAllExceptMermaid();
+    } else {
+      window.addEventListener('load', function() {
+        highlightAllExceptMermaid();
+      });
+    }
```

Comment:
Could you add a comment explaining what `window.isHotReload` does?

Reply:
I'm now less sure of what this is achieving When I tried `bin/run` locally, the old problem (of syntax highlighting not being applied in some cases when switching routes, in live server) remains.

Reply:
OK I've finally found the cause here. The `morphdom` from Ema is removing the `<script async src=".../mermaid.js">` tags.

When the `highlightElement` is called, the `autoloader` plugin adds an async script tag to the `<body>` and sets a callback on its `onload`.

But here a race condition comes in. I'm only guessing here, but it seems like either:
- (happy path) the `<script>` is loaded and `onload` is called before it's removed from `<body>`,
- or the `<script>` is removed before the browser even starts loading the script resulting in no outcomes (no errors logs, and no syntax highlighting),
- or the browser starts fetching the JS file and at the point the fetch completes the `<script>` tag has already been removed resulting in code error as the autoloader plugin tries to remove the `<script>` tag itself when it no longer exists.

I've tried to combat this, and the code grew. Not my proudest work at all. What do you think?

It's a little glitchy when first opening a page, but it at least resolves into a syntax highlighted page eventually. And there's no glitchyness after a hot reload or after just loading the statically generated site.

Reply:
Do you think this whole debounce logic is worth upstreaming to Ema? cf. https://github.com/EmaApps/ema/issues/69

ie., do we envision doing this for anything other than PrismJS?

Reply:
In particular, perhaps we can abstract this out in a function (in Ema) and call it in one line as:

```javascript
Ema.invoke(1000, highlightAllExceptMermaid);
```

Reply:
Adding a debounce function to upstream Ema library just feels like adding it for the sake of it. I'd vote for no as it's quite the simple function.

Anyway that decision is not written in stone. Could extract it first when a need for it appears in multiple places.

Besides, this code is actually me working around Ema's behavior, but perhaps that could be modified instead. Like when using `emanote run` then on a fresh page load it will:

1. run scripts (on regular page load)
2. run scripts when websocket connected
3. load page again (via websockets) and morph DOM
4. run script after DOM morph

I see you have comments in the JavaScript in the sorts of "`We have to reload <script>'s here on initial page load here, so as to make Twind continue to function`", and I don't know the full depth of this, but I think the best option would be if scripts only ran once and morphing the DOM only happens on page changes and not on initial load.